webapp/backend/views.py

```diff
@@ -408,6 +412,129 @@ def withdraw(self, request, uuid, format=None):
+    def stop(self, request, uuid, format=None):
+        """
+        This method is used to stop an application on a specified lambda instance. Responds
+        to POST requests on url r'^{prefix}/{lookup}/{methodname}{trailing_slash}$'
+        """
+
+        application, lambda_instance, instanceapplication = \
+            self.get_application_instance_connection(request, uuid)
+
+        if not instanceapplication.started:
+            raise CustomAlreadyDoneError(CustomAlreadyDoneError.
+                                         messages['application_already_stopped'])
+
+        if application.type == application.BATCH:
+            if not lambda_instance.started_batch:
+                raise CustomAlreadyDoneError(CustomAlreadyDoneError.
+                                             messages['batch_job_already_stopped'])
+            app_type = 'batch'
+        else:
+            if not lambda_instance.started_streaming:
+                raise CustomAlreadyDoneError(CustomAlreadyDoneError.
+                                             messages['streaming_job_already_stopped'])
+            app_type = 'streaming'
+
+        lambda_instance_uuid = lambda_instance.uuid
+        app_action = 'stop'
+
+        # Create a task to stop the application.
+        tasks.start_stop_application.delay(lambda_instance_uuid=lambda_instance_uuid,
+                                           app_uuid=uuid, app_action=app_action, app_type=app_type)
+
+        # Return an appropriate response.
+        status_code = status.HTTP_202_ACCEPTED
+        return Response({
+            "status": {
+                'code': status_code,
+                'short-description': ResponseMessages.short_descriptions['application_stop']
+            }}, status=status_code)
+
+    def get_application_instance_connection(self, request, uuid):
+        """
+        This method is used to withdraw an application from a specified lambda instance. Responds
+        to POST requests on url r'^{prefix}/{lookup}/{methodname}{trailing_slash}$'
+        """
+
+        application_uuid = uuid
+
+        # Check if the lambda instance id was provided with the request.
+        lambda_instance_uuid = request.data.get('lambda_instance_id')
+        if not lambda_instance_uuid:
+            raise CustomParseError(CustomParseError.messages['no_lambda_instance_id_error'])
+
+        # Check if the specified lambda instance exists.
+        lambda_instances = LambdaInstance.objects.filter(uuid=lambda_instance_uuid)
+        if not lambda_instances.exists():
+            raise CustomNotFoundError(CustomNotFoundError.messages['lambda_instance_not_found'])
+        lambda_instance = lambda_instances[0]
+
+        # Check if the specified application exists.
+        applications = self.get_queryset().filter(uuid=application_uuid)
+        if not applications.exists():
+            raise CustomNotFoundError(CustomNotFoundError.messages['application_not_found'])
```

Comment:
There are 2 types of messages, error messages and response messages. Error messages are returned via exceptions. There is a number of exception types each with its own messages. When throwing a custom exception with a message, make sure you add that message to the `messages` dictionary of the respective exception. ( https://github.com/grnet/okeanos-LoD/blob/devel/webapp/backend/exceptions.py )